### ISSUE TITLE
Support other vagrant providers

### ISSUE THREAD

**Issue by matbaj** [koding/koding#9230]
Support other vagrant providers

## Expected Behavior
Support other vagrant providers than Virtualbox like libvirt.

## Current Behavior
koding stacks doesn't support other vagrant providers Virtualbox

## Possible Solution
Implement other vagrant providers support.

## Steps to Reproduce (for bugs)
Setup libvirt provider and add it as vagrant default provider.
Koding is still using Virtualbox with VT support.

## Context
In case if you don't have full physical machine for vagrant virtualbox environment and only VM from openstack, xen, libvirt and you would like to setup koding vagrant enviroment then you are not able to do this as VirtualBox doesn't support VT-nested as host and as a guest and this is somehow required by default from Vagrant to use VT enabled vms.

## Your Environment
* Version used: N/A
* Environment name and version (e.g. Chrome 39, node.js 5.4): N/A
* Operating System and version (desktop or mobile): Ubuntu 16.04 with vagrant libvirt provider
* Link to your project: N/A

**Comment by rjeczalik** [comment]
Currently we use virtualbox as a vagrant provider, the support is hardcoded in a few places:

- vagrantutil (https://github.com/koding/vagrantutil)
- terraform plugin for vagrant (https://github.com/koding/koding/tree/master/go/src/koding/kites/terraformplugins/vagrant)
- vagrantfile template (https://github.com/koding/koding/blob/master/go/src/koding/klient/vagrant/template.go)
- klient vagrant checks (we check for virtualbox kernel module, if present we assume we're in a vagrant stack - if koding is deployed in vagrant this will give false positives)

In order to get there we'd need:

- ensure vagrantutil works with all providers (and solve related issues - https://github.com/koding/koding/issues/9120 and https://github.com/koding/koding/issues/9040)
- make it possible to directly manipulate with vagrantfile from stack template instead of template the hardcoded one
- adapt klient / kd accordingly

Basically the above is a quick TODO list off the top of my head.
